### PR TITLE
Jakarta Security 2.0

### PR DESCRIPTION
**Pull request by jeanouii**
Signed-off-by: Jean-Louis Monteiro <jlmonteiro@tomitribe.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [ ] Email to PMC
   - [ ] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/security/enterprise/jakarta.security.enterprise-api/2.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-security-tck-2.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/security-api/issues/177
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

File: security/2.0/_index.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+date: 2020-09-02
 summary: "Release for Jakarta EE 9"
 ---
 Jakarta Security defines a standard for creating secure Jakarta EE applications in modern application paradigms.
 
 * [Jakarta Security 2.0 Release Record](https://projects.eclipse.org/projects/ee4j.es/releases/2.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Security 2.0 Specification Document](./security-spec-2.0.0-SNAPSHOT.pdf) (PDF)
-* [Jakarta Security 2.0 Specification Document](./security-spec-2.0.0-SNAPSHOT.html) (HTML)
+* [Jakarta Security 2.0 Specification Document](./jakarta-security-spec-2.0.0-SNAPSHOT.pdf) (PDF)
```
Comment (arjantijms) [inline security/2.0/_index.md:10]: The docs that are generated are called `jakarta-security-spec-2.0.0.pdf` and `security-spec-2.0.0.html`, so without the jakarta prefect and without snapshot.

Comment (jeanouii) [reply security/2.0/_index.md]: I have renamed the files and make sure the links were correct and without snapshot.
Now we need to generate the apidocs and spec docs and update the PR.

Do you want me to do that?

File: security/2.0/_index.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+date: 2020-09-02
 summary: "Release for Jakarta EE 9"
 ---
 Jakarta Security defines a standard for creating secure Jakarta EE applications in modern application paradigms.
 
 * [Jakarta Security 2.0 Release Record](https://projects.eclipse.org/projects/ee4j.es/releases/2.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Security 2.0 Specification Document](./security-spec-2.0.0-SNAPSHOT.pdf) (PDF)
-* [Jakarta Security 2.0 Specification Document](./security-spec-2.0.0-SNAPSHOT.html) (HTML)
+* [Jakarta Security 2.0 Specification Document](./jakarta-security-spec-2.0.0.pdf) (PDF)
```
Comment (arjantijms) [inline security/2.0/_index.md:10]: There's still a problem, "jakarta" needs to be removed as well.

File: security/2.0/_index.md (hunk 1 above)
Comment (arjantijms) [reply security/2.0/_index.md]: Yes, sure ;) It's just a matter of running `mvn clean install -Dstatus=Final` in the project, and setting the version to 2.0.

File: security/2.0/_index.md (hunk 2 above)
Comment (arjantijms) [reply security/2.0/_index.md:10]: Oh, and sorry, 2.0.0 needs to become 2.0.

Comment (jeanouii) [reply security/2.0/_index.md:10]: Ok my bad I misunderstood.
Bean Validation has the jakarta in the name of the file. Same for CDI which I tried to use for consistency.
Happy to yank jakarta- prefix from file name here or to update the Jakarta Security project to add it.

What do you prefer?

I'll update the version at the same time.

File: security/2.0/_index.md (hunk 1 above)
Comment (jeanouii) [reply security/2.0/_index.md]: I'll do then

File: security/2.0/_index.md (hunk 2 above)
Comment (jeanouii) [reply security/2.0/_index.md]: For the moment, I removed the jakarta prefix here. We can update docs and link later if we want consistency

Comment (arjantijms) [reply security/2.0/_index.md]: I spot checked a few specs: authentication, authorization, security and expression language, server pages, standard tag library and servlet, but none of these have the jakarta- prefix. So for consistency I think it's best not to have it here either.

Comment (jeanouii) [reply security/2.0/_index.md]: Then all fine. I removed it

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by jeanouii** [comment]
@arjantijms Added back the jakarta- prefix. It is required from now on. Check out the template of the PR which has been updated accordingly.

For Jakarta EE 9, it's only optional for specifications under vote so we don't have to re-roll.
So I've sent a new PR to security project and also updated this one.

**Comment by arjantijms** [comment]
@jeanouii okay, that’s fine. Then we need to update the specs not under vote as well to update this.

**Comment by starksm64** [comment]
@arjantijms is there a TCK coming soon?

**Comment by arjantijms** [comment]
> is there a TCK coming soon?

Yes, this should be coming soon indeed. I'll take a look tomorrow.

**Comment by starksm64** [comment]
Scotts-iMacPro:tmp starksm$ shasum -a256 jakarta-security-tck-2.0.0.zip 
63563b05a7b09d7d840395570f4466e129026dfd071bd037ef159a9f817bf338  jakarta-security-tck-2.0.0.zip

**Comment by arjantijms** [comment]
TCK executed against staged API jar and stages Soteria using the latest GlassFish weekly build:

```
[javatest.batch] Number of Tests Passed      = 85
[javatest.batch] Number of Tests Failed      = 0
[javatest.batch] Number of Tests with Errors = 0
[javatest.batch] ********************************************************************************

[javatest.batch] Total time = 175s
[javatest.batch] Setup time = 0s
[javatest.batch] Cleanup time = 0s
[javatest.batch] Test results: passed: 85

SHA256_GF=f69a582e2ee15f49afb48adb2393d42351256ea070a60513e1ae17afe973e9c3
SHA256_API=4543f3b397ef9d88c8d915cd398424670b19df8bf97d3927efcaed6f5eb9415a
SHA256_TCK=63563b05a7b09d7d840395570f4466e129026dfd071bd037ef159a9f817bf338
SHA256_IMPL=c2fef7dacd12de3506ee24311ea9653102faf2645dbde9f966385dfbcddd10fc
SHA256_TCK=63563b05a7b09d7d840395570f4466e129026dfd071bd037ef159a9f817bf338

Product_download=https://download.eclipse.org/ee4j/glassfish/weekly/glassfish-6.0.0-SNAPSHOT-2020-10-04.zip
IMPL_download=https://jakarta.oss.sonatype.org/content/repositories/staging/org/glassfish/soteria/jakarta.security.enterprise/2.0.0-M3/jakarta.security.enterprise-2.0.0-M3.jar
TCK_download=https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-security-tck-2.0.0.zip

OS4=3.12.0 NAME="Alpine Linux" ID=alpine VERSION_ID=3.12.0 PRETTY_NAME="Alpine Linux v3.12" HOME_URL="https://alpinelinux.org/" BUG_REPORT_URL="https://bugs.alpinelinux.org/"
JDK_VERSION=java version "1.8.0_202" Java(TM) SE Runtime Environment (build 1.8.0_202-b08) Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
```

**Comment by arjantijms** [comment]
Compatibility certification request: https://github.com/eclipse-ee4j/security-api/issues/177

**Comment by arjantijms** [comment]
Public TCK results: https://eclipse-ee4j.github.io/soteria/certifications/jakarta-security/2.0/TCK-Results

**Comment by arjantijms** [comment]
CQ / IP Log bug: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22692

**Review by starksm64 — APPROVED** [review]
I have updated the final TCK locations on the _index.md page, so all that is left to do are the EMO/PMC tasks, which can be done in parallel with the specification committee ballot.

**Comment by starksm64** [comment]
Ballot thread
https://www.eclipse.org/lists/jakarta.ee-spec/msg01004.html

**Comment by jeanouii** [comment]
Very good job guys.

**Comment by arjantijms** [comment]
TCK promotion issue: https://github.com/eclipse-ee4j/jakartaee-tck/issues/530

**Comment by kwsutter** [comment]
@starksm64 It looks like your commits were not signed...  Can you take care of that before you do the merge?  Thanks!

**Comment by kwsutter** [comment]
In the Spec "front matter", September looks like it might be the French spelling?  Don't sweat it for this ballot, but you should probably fix this for the next release.  Thanks.

Specification: Jakarta Security
Version: 2.0
Status: Final
Release: septembre 03, 2020

**Review by kwsutter — CHANGES_REQUESTED** [review]
@arjantijms and @jeanouii, we need a more specific reference for the Compatible Implementation.  There is a reference to an RC1 for Soteria 2.0.0, but I can't find that anywhere.  Even the tags on the project don't reference an RC1 (https://github.com/eclipse-ee4j/soteria/tags).  This is consistent across this PR, the CR, and the TCK results.  We need a specific version of Soteria that was used to test the TCK and the Spec.  This needs to be resolved before this PR can be merged (but I'll still vote for the ballot).

**Comment by arjantijms** [comment]
@kwsutter 
There's an, admittedly not very obvious, reference to the implementation that was used to run the TCK here:

https://eclipse-ee4j.github.io/soteria/certifications/jakarta-security/2.0/TCK-Results

It's near the bottom:

```
IMPL_download=https://jakarta.oss.sonatype.org/content/repositories/staging/org/glassfish/soteria/jakarta.security.enterprise/2.0.0-M3/jakarta.security.enterprise-2.0.0-M3.jar
```

I'll update it where I can and will create the release tag with attached binaries.

**Comment by arjantijms** [comment]
I updated the CR and the TCK results. @jeanouii will have to update the PR, as I don't have access to that.

**Comment by starksm64** [comment]
* On ballot completion, the specification committee mentor:
 - [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>

**Pull request merged**